### ISSUE TITLE
sudo in a local_resource breaks TUI rendering and input

### ISSUE THREAD

**Issue by landism** [tilt-dev/tilt#2982]
### Repro

Put this in a Tiltfile and `tilt up`
`local_resource("foo", "sudo echo foo")`

### Observed

The password prompt overwrites the UI, and gets input into a totally confusing state (e.g., I hit "enter" after typing and it opens the browser, I try to ctrl+C to get out of Tilt, and it doesn't work):

![image](https://user-images.githubusercontent.com/7453991/74854400-4a08ea80-530d-11ea-8abd-4c2456333459.png)

### Expected

I'm not entirely sure what a good behavior here is.
Options that come to mind:
* make sure that Tilt execs processes with no TTY, so that sudo can't screw up the Tilt TTY, and hopefully the Tilt logs end up with a nice error from sudo saying there's no TTY, so at least users know what's happening and can try to find a workaround
* set SUDO_ASKPASS when running local processes and ship with something appropriate to set it to

**Comment by landism** [comment]
> set SUDO_ASKPASS when running local processes and ship with something appropriate to set it to

Another option would be to set SUDO_ASKPASS to something like /sudo/not/supported/in/tilt, to get an acceptable error message without screwing up the terminal. However, at least on MacOS, it turns out sudo only even checks SUDO_ASKPASS if you call it with -A, so any solutions involving SUDO_ASKPASS are probably out.

**Comment by omh1280** [comment]
Is there any progress or any other potential workarounds for this problem? We're trying to run [`kubefwd`](https://github.com/txn2/kubefwd) and it requires sudo

**Comment by landism** [comment]
A user just reported this issue on Slack and I repro'd.

Repro:
With this Tiltfile
```
local("sudo ls")
local_resource('blah', 'echo blah')
```

`tilt up` and get this:
```
Tilt started on http://localhost:10350/
v0.22.6-dev, built 2021-09-01

(space) to open the browser
(s) to stream logs (--stream=true)
(t) to open legacy terminal mode (--legacy=true)
(ctrl-c) to exit
Password:
```

If you hit the 's' key a few times, eventually the Tilt prompt will catch it and start streaming logs and enable echo, and sudo will still be taking input.
When you then ctrl+C tilt, echo will be disabled (i.e., you can't see what you type).

NB: the user on slack experienced this from the git_resource extension trying to clone via ssh, and their password contained a 't', which opened the TUI, which is more awkward to type into.

**Comment by nicks** [comment]
eh....i think that's a much different issue. you can read more about it here:
https://github.com/tilt-dev/tilt-extensions/issues/239

if a tool isn't properly doing TTY detection, then there's no real way (at the OS level) to prevent it from doing this. 

I'll file a separate bug against the git_resource extension to use the right env variables. (here's how the golang project handles this: https://github.com/golang/go/blob/acdea4f9f792339ee27fb1f67539a7ca2e3af99a/src/cmd/go/internal/get/get.go#L135)

**Comment by nicks** [comment]
(and TBH i thought we were already setting the right env variables, so maybe there's a deeper issue here)

**Comment by landism** [comment]
> eh....i think that's a much different issue. you can read more about it here:
> tilt-dev/tilt-extensions#239

The original issue was "sudo breaks stuff" and it was closed with a comment saying it no longer repros, and I thought I repro'd it, so I figured it was fair to reopen. I hadn't realized it repro'd with local but not local_resource. I was thinking of this as a general "Tilt gets broken by its subprocesses messing with the tty" issue, but maybe that was unfair.

> if a tool isn't properly doing TTY detection, then there's no real way (at the OS level) to prevent it from doing this.

Sounds fair. That's a clear resolution!

> We run local resources without a TTY

I was surprised this repros for me with `local` but not `local_resource` - I don't see a difference in their code path that'd explain this. But it's probably not worth spending more time on since that's not actually an issue a user hit.
